uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestTest.php

```diff
@@ -115,38 +115,32 @@ public function testCreateDoesNotDuplicateLeaveBalanceChangesOnUpdate() {
     $this->assertCount(11, $balanceChanges);
   }
 
-  public function testDeleteDeletesTheLeaveRequestItsBalanceChangesAndDates() {
+  public function testDeleteSoftDeletesTheLeaveRequestAndThatFetchingItsBalanceChangesAndDatesReturnsEmpty() {
```

Comment:
This test looks very weird to me. The only thing that the delete() method does is call the softDelete() method on the BAO, so that should be the only thing being tested here. There should be only a single assert to check if the is_deleted value was updated to 1. The method name should not mention anything about dates or balance changes (it was ok before, because the previous version also deleted these things)